### ISSUE TITLE
consider changing data logic

### ISSUE THREAD

**Issue by sgratzl** [cmu-delphi/forecast-eval#174]
atm. the app loads all the data stored in multiple files (cases, death, hospitalizations, per us/states) and then one of the first steps is to filter them again by targetVariable (cases, deaths, ...), scoreType, and location. 

one option would be to load only the data that is really needed and better split the up in multiple files (targetVariable x score x location (nation or states)). This would reduce the initial loading time and with https://shiny.rstudio.com/reference/shiny/1.6.0/bindCache.html shiny could take care of caching datasets.

**Comment by ryantibs** [comment]
Thanks @krivard.  And sounds like a good idea, @sgratzl 

What happened to the idea of reading data from disk instead of from the S3 bucket?  I remember @nmdefries mentioning that it's actually slow to read from the S3 bucket.  And then I suggested we just download a local clone of all the data each Monday that we can read from for the dashboard.  Has that been implemented and does it lead to speed improvements?

**Comment by nmdefries** [comment]
I've implemented caching for the score data in #169 that only loads the score data twice a week, after each pipeline run. Every other user besides the first one following a pipeline run will be reading the scores from memory. Releasing the caching change is waiting on another PR -- let me go poke some people.

**Comment by nmdefries** [comment]
This behavior has been added for target variables, but geo types are still combined due to the only small additional benefit that change would provide. Most of the time we're handling state forecasts, both when a single state is requested or when we're summarizing across them; separating out US data only saves effort ~1/60 of the time.
